src/NServiceBus.Core/MessageInterfaces/MessageMapper/Reflection/MessageMapper.cs

```diff
@@ -240,24 +215,21 @@ void GenerateImplementationFor(Type interfaceType)
     static string GetTypeName(Type t)
     {
         var args = t.GetGenericArguments();
-        if (args.Length == 2)
+        if (args.Length != 2)
```

Comment:
I don't really feel like the changes here are an improvement over what was here before. You're now having to duplicate the `FullName` case...

Reply:
Feel free to change back